clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerMembershipManager.java

```diff
@@ -188,6 +186,7 @@ public ConsumerMembershipManager(String groupId,
         this.rebalanceTimeoutMs = rebalanceTimeoutMs;
         this.serverAssignor = serverAssignor;
         this.commitRequestManager = commitRequestManager;
+        this.commitRequestManager.memberInfo.memberId = this.memberId;
```

Comment:
This line will cause tests to fail in `ConsumerMembershipManagerTest` because `commitRequestManager` is a mock, meaning its `memberInfo` is never initialized. As a result, assigning a value will lead to a NPE. 

I'm considering how to handle this issue.

Reply:
well from your [comment](https://github.com/apache/kafka/pull/17549#discussion_r1815213715) above it seems that even if we set the member ID explicitly, it is still needed to keep it on the `onMemberEpochUpdated` as param right? if so then it's probably not worth the hassle, we can just remove this line and stay with the memberID updated on the listeners as part of the onMemberEpochUpdated